ktor-io/jvm/src/io/ktor/utils/io/ByteBufferChannel.kt

```diff
@@ -1631,6 +1631,7 @@ internal open class ByteBufferChannel(
 
     override suspend fun read(min: Int, consumer: (ByteBuffer) -> Unit) {
         require(min >= 0) { "min should be positive or zero" }
+        if (min == 0) return
```

Comment:
is it really correct?
f.e. if we call it with `min=0` and we have 5 bytes in channel, I think, that those should be consumed, no?
And in case there will be no bytes in channel I don't know what is expected behaviour - may be return or call consumer with empty buffer, any way, no suspension should happen to await more bytes.
So in my opinion, it should work like non-suspend `readAvailable`, no?

Reply:
Thanks, will check